http.go

```diff
@@ -165,26 +165,24 @@ func (g *Gateway) GraphQLHandler(w http.ResponseWriter, r *http.Request) {
 	emitResponse(w, statusCode, string(response))
 }
 
-// Parses request to operations (single or batch mode)
-func parseRequest(r *http.Request) (operations []*HTTPOperation, batchMode bool, payloadErr error) {
+// Parses request to operations (single or batch mode).
+// Returns an error and an error status code if the request is invalid.
+func parseRequest(r *http.Request) (operations []*HTTPOperation, batchMode bool, errStatusCode int, payloadErr error) {
 	// this handler can handle multiple operations sent in the same query. Internally,
-	// it modules a single operation as a list of one.
+	// it models a single operation as a list of one.
 	operations = []*HTTPOperation{}
-
-	// the error we have encountered when extracting query input
-
-	// make our lives easier. track if we're in batch mode
-	batchMode = false
```

Comment:
Removed this line since it is already false by default when not set. (Go's zero-value)